azure/dns-records/variables.tf

```
@@ -0,0 +1,24 @@
+variable "dns_zone" {
+  description = "Name of the DNS zone"
+  type        = string
+}
+
+variable "dns_rg" {
+  description = "Resource Group of the DNS Zone"
+  type        = string
+}
+
+variable "a" {
+  description = "A record to be created"
+  type        = map(any)
+}
+
+variable "cname" {
+  description = "CNAME record to be created"
+  type        = map(any)
+}
+
+variable "txt" {
+  description = "TXT record to be created"
+  type        = map(any)
```

It might be a good idea to specify the exact type of the map, so that it's easier to spot a mismatch between the expected type and the variable in the service, or else the developer needs to look through the module's code or find an existing implementation to use as an example.

We could also set a default value using an empty object (default { }) to make the map optional. 
```suggestion
  type = map(object({
    text    = string
    records = list(string)
  }))
  default = {}
```